src/binary/ibc-setup/commands/init.ts

```diff
@@ -0,0 +1,57 @@
+import fs from 'fs';
+import path from 'path';
+
+import { Command } from 'commander';
+import yaml from 'js-yaml';
+
+import { GlobalOptions } from '../types';
+import { resolveOption } from '../utils/resolve-option';
+
+type Options = GlobalOptions & {
+  src: string;
+  dest: string;
+};
+
+type ProgramOptions = Partial<Options>;
+
+function getDefaultHome() {
+  if (!process.env.HOME) {
+    throw new Error('$HOME environment variable is not set.');
+  }
+
+  return `${process.env.HOME}/.ibc-setup`;
+}
+
+const REGISTRY_FILE = 'registry.yaml';
+
+export function init(_: unknown, program: Command) {
+  const programOptions: ProgramOptions = program.opts();
+
+  const options = {
+    src: resolveOption('src', programOptions.src, process.env.RELAYER_SRC),
+    dest: resolveOption('dest', programOptions.dest, process.env.RELAYER_DEST),
+    home: resolveOption('home', programOptions.home, getDefaultHome),
+  };
+
+  run(options);
+}
+
+export function run(options: Options) {
+  if (!fs.existsSync(options.home)) {
+    fs.mkdirSync(options.home, { recursive: true });
+    console.log(`Initialized home directory at ${options.home}`);
+  } else if (!fs.lstatSync(options.home).isDirectory()) {
+    throw new Error(`${options.home} must be a directory. It is a file.`);
+  }
+
+  const REGISTRY_FILE_PATH = path.join(options.home, REGISTRY_FILE);
+  if (!fs.existsSync(REGISTRY_FILE_PATH)) {
+    // TODO: download registry.yaml from default location
+  } else if (!fs.lstatSync(REGISTRY_FILE_PATH).isFile()) {
+    throw new Error(`${REGISTRY_FILE_PATH} must be a file. It is a directory.`);
+  }
+
+  const registry = yaml.load(fs.readFileSync(REGISTRY_FILE_PATH, 'utf-8'));
+  // TODO: registry file validation?
```

Comment:
Should we care about the registry validation or skip it for now?

It'll be sort-of validated in the succeeding functions. But maybe we should validate it and cast it to concrete type? Thoughts?

Reply:
It would be nice to assert a type and validate it.
Again, let's decide if we take this route and add later

Reply:
Alright.

I think no matter of route we would have to do it.
4b45fc2